lib/_stream_readable.js

```diff
@@ -526,6 +537,14 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     }
   }
 
+  // if src has an error and error forwarding is turned on
+  // cause the error to be emited in dest and do suppress the
+  // throwing behavior
+  function onsrcerror(er) {
+    debug('onsrcerror', er);
+    dest.emit('error', er);
```

Comment:
But the writable half could forward to the readable half

On Sat, Apr 25, 2015, 8:49 PM Domenic Denicola <notifications@github.com>
wrote:

> In lib/_stream_readable.js
> <https://github.com/iojs/io.js/pull/1521#discussion_r29105716>:
>
> > @@ -526,6 +537,14 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
> >      }
> >    }
> >
> > +  // if src has an error and error forwarding is turned on
> > +  // cause the error to be emited in dest and do suppress the
> > +  // throwing behavior
> > +  function onsrcerror(er) {
> > +    debug('onsrcerror', er);
> > +    dest.emit('error', er);
>
> An object cannot be both readable and writable so the problem is avoided.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iojs/io.js/pull/1521/files#r29105716>.
>